commit 15e230abaaa5a79feb25d1ef29b380e8b63aea0c
Author: Fred Zhou <fred.zy@gmail.com>
Date:   Tue Sep 24 10:33:01 2013 +0800

    mac80211: use exact-size allocation for authentication frame
    
    The authentication frame has a fixied size of 30 bytes
    (including header, algo num, trans seq num, and status)
    followed by a variable challenge text.
    Allocate using exact size, instead of over-allocation
    by sizeof(ieee80211_mgmt).
    
    Signed-off-by: Fred Zhou <fred.zy@gmail.com>
    Signed-off-by: Johannes Berg <johannes.berg@intel.com>

diff --git a/net/mac80211/util.c b/net/mac80211/util.c
index 3c8283b9b45a..1fb31f94bf89 100644
--- a/net/mac80211/util.c
+++ b/net/mac80211/util.c
@@ -1081,8 +1081,8 @@ void ieee80211_send_auth(struct ieee80211_sub_if_data *sdata,
 	struct ieee80211_mgmt *mgmt;
 	int err;
 
-	skb = dev_alloc_skb(local->hw.extra_tx_headroom +
-			    sizeof(*mgmt) + 6 + extra_len);
+	/* 24 + 6 = header + auth_algo + auth_transaction + status_code */
+	skb = dev_alloc_skb(local->hw.extra_tx_headroom + 24 + 6 + extra_len);
 	if (!skb)
 		return;